openedx_atlas/sparse_checkout.py

```diff
@@ -0,0 +1,40 @@
+"""
+Performs sparse-checkout of subdirectories for a given <repository>:<branch>.
+
+Contains one function: sparse_checkout.
+"""
+import os
+import tempfile
+from shutil import copytree, ignore_patterns
+from subprocess import call
+
+
+def sparse_checkout(directory, repository, branch, output_path=os.getcwd()):
+    """
+    Run a git sparse-checkout for a given repository, branch, and directory.
+    """
+    # Make a temporary directory
+    with tempfile.TemporaryDirectory() as tmp_dir_name:
+        # set temporary directory as current directory
+        os.chdir(tmp_dir_name)
+        # initialize directory as git repo
+        call("git init", shell=True)
```

Comment:
All of these calls should be getting run in an already code-reviewed context, so the yml or inputs should be defined in places that have been reviewed before these are called. I can't think of a way where someone would be able to add malicious arguments in the intended use of this. 👍